prejpms/src/main/java/com/baeldung/prejpms/App.java

```diff
@@ -0,0 +1,79 @@
+package com.baeldung.prejpms;
+
+import java.io.StringWriter;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.crypto.provider.SunJCE;
+
+import sun.misc.BASE64Encoder;
+import sun.reflect.Reflection;
+
+public class App {
+
+    private static final Logger logger = LoggerFactory.getLogger(App.class);
+
+    public static void main(String[] args) throws Exception {
+
+        getCrytpographyProviderName();
+        getCallStackClassNames();
+        getXmlFromObject(new Book(100, "Java Modules Architecture"));
+        getBase64EncodedString("Java");
+    }
+
+    private static void getCrytpographyProviderName() {
+        try {
+            logger.info("1. Java Cryptography Extension - Provider Name: " + new SunJCE().getName() + "\n");
+        } catch (Throwable e) {
+            logger.error(e.toString());
+        }
+    }
+
+    private static void getCallStackClassNames() {
+        try {
```

Comment:
For the purposes of separating the callstack walker from the boilerplate, can I suggest you extract a function that does the insides of this try block up to the logger statement, and returns a String.

Reply:
Extracting a function will make the caller a single line function excluding the try catch block. And in the article, we may have to show both the functions instead of one to connect the dots.
I've explained why I went for individual try-catch blocks in below comment. Please let me know if you feel I should still go ahead.

Reply:
I'm happy with that.